src/machine/machine_atsamd21.go

```diff
@@ -1910,12 +1910,12 @@ func (f flashBlockDevice) EraseBlocks(start, len int64) error {
 
 // pad data if needed so it is long enough for correct byte alignment on writes.
 func (f flashBlockDevice) pad(p []byte) []byte {
-	paddingNeeded := f.WriteBlockSize() - (int64(len(p)) % f.WriteBlockSize())
+	paddingNeeded := int64(len(p)) % f.WriteBlockSize()
```

Comment:
If I understand correctly, this is not the padding needed, but 'the amount of data that partially fills the last block' (the inverse of the padding needed).

Perhaps a different name would be better - like 'spillover'?

Reply:
Good idea to rename it. I went with `overflow`.